zephyr/wrapper.c

```diff
@@ -98,10 +74,10 @@ void *rmalloc(enum mem_zone zone, uint32_t flags, uint32_t caps, size_t bytes)
 		return NULL;
 	}
 
-	new_ptr = (void *)((unsigned long)ptr + 32);
+	new_ptr = (void *)((unsigned long)ptr + sizeof(struct __alloc_hdr));
 	*((void **)new_ptr - 1) = ptr;
 
-	alloc_table_add(new_ptr);
+	sys_slist_append(&alloc_list, &((struct __alloc_hdr *)ptr)->snode);
```

Comment:
Are these primitives concurrency-aware? Or memory allocations themselves protected by some spinlock?

Reply:
The memory allocation is protected by spinlocks but that slist is needed for debug only, I might add locking.

Reply:
No need if the allocation itself is protected, thanks for mentioning.